### ISSUE TITLE
Improve the inputs `help` prop documentation 

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#63692]
### Description

Splitting this out from https://github.com/WordPress/gutenberg/issues/61196 / https://github.com/WordPress/gutenberg/pull/63669

Most (all?) of the input controls that extend the input base component accept a `help` prop. This prop is then used to automatically generate an element containg a description that is then associated via an `aria-describedby` attribute to the input. So far so good.

However, it appears the expected usage of this prop isn't that clear to contributors. For example, in [the PostURL component](https://github.com/WordPress/gutenberg/blob/6b3cb26a7c9e11efcd69d78872466d69ad2389fc/packages/editor/src/components/post-url/index.js#L143-L159) the help prop is being passed an External link component that renders a full link with the additional text '(opens in a new tab)'. That's arguably a meaningful description for the input field to edit the post _slug_.

For example, that input field is exposed to assistive technologies this way:
- accessible name: `Link`
- accessible description: `http://mydomain.org/my-awesome-and-long-post-slug/ (opens in a new tab)`

I'm guessing this External link has been used as the `help` prop only for visual purposes, to place the link below the input. That's less than ideal.

The `help` prop should be used for:
- either a meaningful description of the input
- or for instructions about what to input

As a developer, when hovering the `help` prop in my editor, I do get a popup with the `help` prop documentation. Apparently, this isn't sufficient to help contributors fully understand the correct usage of this prop. Screenshot:

![Screenshot 2024-07-18 at 08 52 42](https://github.com/user-attachments/assets/9a69d43a-6323-4740-88bf-028d82ee0fda)

I would like to propose to update this prop documentation and clearly explain it must be used only for meaningful descriptions or instructions.

See the base control:
- [readme](https://github.com/WordPress/gutenberg/blob/6b3cb26a7c9e11efcd69d78872466d69ad2389fc/packages/components/src/base-control/README.md#L54-L59)
- [types.ts file](https://github.com/WordPress/gutenberg/blob/6b3cb26a7c9e11efcd69d78872466d69ad2389fc/packages/components/src/base-control/types.ts#L21-L26)

### Step-by-step reproduction instructions

N/A

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [X] Yes
